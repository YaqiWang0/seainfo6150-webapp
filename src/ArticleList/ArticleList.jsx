import React from "react";
import ArticleListItem from "../ArticleListItem/ArticleListItem";

const ArticleList = (props) => {
    let displayContent;
    if(props.articles.length) {
        displayContent = (

            <div>
                <ul>
                {props.articles.map((item) => (
                        <ArticleListItem article={item} key={item.slug}/>
                    ))}
                 </ul>
            </div>
        )
    } else {
        displayContent = <div>You have no data!</div>
    }

    return props.articles.length ? (
        <div>
            {displayContent}
        </div>
    ) : null;
};

export default ArticleList;
