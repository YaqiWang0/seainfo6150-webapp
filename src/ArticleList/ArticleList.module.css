@media screen and (min-width: 200px) {
    .container {
        display: grid;
        grid-template-columns: 1fr;
        flex-direction: column;
        background-color: grey;
        box-shadow: 0 0 5px 5px rgba(255, 0, 0, 0.5);
        margin: 15px;
        border-radius: 20px;
    }
}

@media screen and (min-width: 550px) {
    .container {
        display: grid;
        grid-template-columns: 1fr 1fr;
        flex-direction: column;
        background-color: grey;
        box-shadow: 0 0 5px 5px rgba(255, 0, 0, 0.5);
        margin: 15px;
        border-radius: 20px;
    }
}

@media screen and (min-width: 900px) {
    .container {
        display: grid;
        grid-template-columns: 1fr 1fr 1fr;
        flex-direction: column;
        background-color: grey;
        box-shadow: 0 0 5px 5px rgba(255, 0, 0, 0.5);
        margin: 15px;
        border-radius: 20px;
    }
}

@media screen and (min-width: 1500px) {
    .container {
        display: grid;
        grid-template-columns: 1fr 1fr 1fr 1fr;
        flex-direction: column;
        background-color: grey;
        box-shadow: 0 0 5px 5px rgba(255, 0, 0, 0.5);
        margin: 15px;
        border-radius: 20px;
    }
}

