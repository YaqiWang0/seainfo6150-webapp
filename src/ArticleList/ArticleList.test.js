import React from "react";
import ArticleList from "./ArticleList";
import data from "./test"

describe("ArticleList tests", () => {
    it("renders correctly", () => {
        const { container  } = render(<ArticleList articles = {data}/>);
        expect(container ).toMatchSnapshot();
    });
})