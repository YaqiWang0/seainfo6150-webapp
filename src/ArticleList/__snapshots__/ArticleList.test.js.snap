// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <div
        class="container"
      >
        <div>
          <img
            alt="Airline etiquette: Tips from an expert "
            class="image"
            src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
          />
        </div>
        <div
          class="title"
        >
          Airline etiquette: Tips from an expert 
        </div>
        <button
          class="button"
        >
          Show more
        </button>
      </div>
      <div
        class="container"
      >
        <div>
          <img
            alt="The 32 most beautiful places to visit in Ireland"
            class="image"
            src="https://cdn.cnn.com/cnnnext/dam/assets/181122102925-statue-of-liberty-crown-super-169.jpg"
          />
        </div>
        <div
          class="title"
        >
          The 32 most beautiful places to visit in Ireland
        </div>
        <button
          class="button"
        >
          Show more
        </button>
      </div>
    </div>
  </div>
</div>
`;
